### PR TITLE
 Adding Cisco support for the Syslog parser

### PR DESCRIPTION
**Pull request by ph**
Add support for the "sequence" number in the log format send by cisco devices.
The number will be extracted to "event.sequence"

Fixes: #10654

### CODE REVIEW

**Comment by ph** [comment]
@webmat I am currently extracting the sequence number generated from Cisco switch (syslog variant), I've looked at ECS, I saw there [was a proposal](https://github.com/elastic/ecs/issues/129 to add an `event.sequence` to the format but nothing concrete. I think its a good time to add it, WDYT?

File: filebeat/input/syslog/parser.rl [hunk 1]
```diff
@@ -61,6 +63,21 @@ func Parse(data []byte, event *event) {
         event.SetNanosecond(data[tok:p])
       }
 
+      # NOTES: This allow to bail out of obvious non valid
+      # hostname, this might no be ideal in all situation, but
```
Comment (andrewkroh) [inline filebeat/input/syslog/parser.rl]: `s/no/not/`

File: filebeat/input/syslog/parser.rl [hunk 2]
```diff
@@ -61,6 +63,21 @@ func Parse(data []byte, event *event) {
...
+      # when this happen we just go to the catch call case and at least
```
Comment (andrewkroh) [inline filebeat/input/syslog/parser.rl]: `s/call/all/`

File: filebeat/input/syslog/parser_test.go [hunk 3]
```diff
@@ -31,6 +31,25 @@ func TestParseSyslog(t *testing.T) {
 		log    []byte
 		syslog event
 	}{
+		{
+			title: "Cisco's syslog",
+			log:   []byte("<190>589265: Feb 8 18:55:31.306: %SEC-11-IPACCESSLOGP: list 177 denied udp 10.0.0.1(53640) -> 10.100.0.1(15600), 1 packet"),
+			syslog: event{
+				priority:   190,
+				message:    "list 177 denied udp 10.0.0.1(53640) -> 10.100.0.1(15600), 1 packet",
+				hostname:   "%SEC-11-IPACCESSLOGP",
```
Comment (andrewkroh) [inline filebeat/input/syslog/parser_test.go]: Unfortunately this isn't a hostname. It's like a tag for the message type.

Comment (ph) [reply filebeat/input/syslog/parser_test.go]: I fixed it.

**Comment by ph** [comment]
@andrewkroh I've made the changes, I think that should be ok

File: CHANGELOG.next.asciidoc [hunk 4]
```diff
@@ -184,6 +184,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Fix bad bytes count in `docker` input when filtering by stream. {pull}10211[10211]
 - Fixed data types for roles and indices fields in `elasticsearch/audit` fileset {pull}10307[10307]
 - Ensure `source.address` is always populated by the nginx module (ECS). {pull}10418[10418]
+- Add support for Cisco switch log. {pull}10760[10760]
```
Comment (andrewkroh) [inline CHANGELOG.next.asciidoc]: I think it would be good to mention "syslog" here for some context.

File: CHANGELOG.next.asciidoc [hunk 5]
```diff
@@ -81,7 +81,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - The `elasticsearch/deprecation` fileset now indexes the `component` field under `elasticsearch` instead of `elasticsearch.server`. {pull}10445[10445]
 - Remove field `kafka.log.trace.full` from kafka.log fielset. {pull}10398[10398]
 - Change field `kafka.log.class` for kafka.log fileset from text to keyword. {pull}10398[10398]
-- Address add_kubernetes_metadata processor issue where old source field is 
+- Address add_kubernetes_metadata processor issue where old source field is
```
Comment (kaiyan-sheng) [inline CHANGELOG.next.asciidoc]: Is this change intended?

Comment (ph) [reply CHANGELOG.next.asciidoc]: Well, I presume I wont be the only one removing the additional space, I propose we make the change :)

**Comment by ph** [comment]
jenkins test this please

**Comment by webmat** [comment]
Yes, I like this change. I've made a note to introduce officially to ECS. I agree with the datatype and the name.

Go for it!

**Comment by inqueue** [comment]
Hi @ph Is there any chance this can be backported to 6.7?

**Comment by ph** [comment]
@inqueue I can backport it, in retrospect is more of a bug than a new feature.

**Comment by ph** [comment]
@inqueue backport PR at https://github.com/elastic/beats/pull/11977
